client/network/src/config.rs

```diff
@@ -643,6 +731,88 @@ impl NonReservedPeerMode {
 	}
 }
 
+///Doc
+#[derive(Clone, Debug)]
+pub enum PreShareKeyConfig {
+	/// A pre-shared key configuration.
+	PRESHAREDKEY(PreSharedKeySecret),
```

Comment:
Do we need this to be enum? We have only one value which doesn't bring anything. I suggest changing this to name to encryption algorithm we're gonna use, or changing this to simple struct

Reply:
ok